internal/manifests/queryfrontend/query_frontend.go

```diff
@@ -0,0 +1,279 @@
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "query-frontend",
+							Image: "docker.io/grafana/tempo:1.5.0",
+							Args: []string{
+								"-target=query-frontend",
+								"-config.file=/conf/tempo.yaml",
+								"-mem-ballast-size-mbs=1024",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          httpPortName,
+									ContainerPort: portHTTPServer,
+									Protocol:      corev1.ProtocolTCP,
+								},
+								{
+									Name:          grpcPortName,
+									ContainerPort: portGRPCServer,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							// TODO do we need to set resources
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      configVolumeName,
+									MountPath: "/conf",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "data-querier-frontend",
+									MountPath: "/var/tempo",
+								},
+							},
+						},
+						{
+							Name:  "tempo-query",
+							Image: "docker.io/grafana/tempo-query:1.5.0",
+							Args: []string{
+								"--query.base-path=/",
+								"--grpc-storage-plugin.configuration-file=/conf/tempo-query.yaml",
+								"--query.bearer-token-propagation=true",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          jaegerUIPortName,
+									ContainerPort: portJaegerUI,
+									Protocol:      corev1.ProtocolTCP,
+								},
+								{
+									Name:          jaegerMetricsPortName,
+									ContainerPort: portQueryMetrics,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							// TODO do we need to define Resources?
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      configVolumeName,
+									MountPath: "/conf",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "data-query",
+									MountPath: "/var/tempo",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: configVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: manifestutils.Name("", tempo.Name),
+									},
+								},
+							},
+						},
+						{
+							Name: "data-querier-frontend",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: "data-query",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err := manifestutils.ConfigureStorage(tempo, &d.Spec.Template.Spec)
+	if err != nil {
+		return nil, err
+	}
+	return d, nil
+}
+
+func services(tempo v1alpha1.Microservices) []*corev1.Service {
+	selectorLabels := manifestutils.ComponentLabels(componentName, tempo.Name) // TODO is there a better way to do this?
+	delete(selectorLabels, "app.kubernetes.io/managed-by")
```

Comment:
why are you deleting these two labels?

Reply:
oh, let's be consistent now across the codebase and use match labels that are used in the other components. 

It's just important to keep the selector immutable to avoid issues like https://github.com/open-telemetry/opentelemetry-operator/pull/1152. Used labels are immutable now, but we could use a smaller set to avoid any potential changes in the future.